Content.Server/GameObjects/Components/Power/ApcNetComponents/PowerProviderComponent.cs

```diff
@@ -59,6 +62,21 @@ public void RemoveReceiver(PowerReceiverComponent receiver)
             Net.UpdatePowerProviderReceivers(this);
         }
 
+        public INodeGroup GetWireNet()
+        {
+            if(Owner.TryGetComponent<NodeContainerComponent>(out var nodeContainer))
+            {
+                var nodes = nodeContainer.Nodes;
+                for (int index = 0; index < nodes.Count; index++)
+                {
+                    if (nodes[index].NodeGroupID == NodeGroupID.WireNet)
+                        return nodes[index].NodeGroup;
+                }
+
+            }
+            return default;
+        }
+
         public override void ExposeData(ObjectSerializer serializer)
```

Comment:
This method doesn't seem to use any private or protected information from this class and is only called in one place, so I don't think this method should be on this class.

Reply:
It looks like you made this because the Owner is hidden behind IPowerProvider, I'd rather you make a property that returns Owner and letting the other interested class take it from there.